### DISCUSSION TITLE
When to see arm64 support on MacOS?

### DISCUSSION THREAD

**Discussion by ssbarnea** [P403n1x87/austin#114] [Q&A] [answered]
When are we going to see arm64 support on MacOS?

**Comment by P403n1x87** [comment, marked as answer]
👋 @ssbarnea despite what the README says, arm64 should be supported. Have you tried installing Austin from homebrew?

**Reply by ssbarnea** [reply]
Yes. I installed it via brew and the executable appears to run but I still get an error when trying to run it (fails to start it).

**Reply by P403n1x87** [reply]
Could you open an issue and share the error message that you're getting? 🙇

**Reply by ssbarnea** [reply]
I did not raise a bug because I did not yet sure on which project it should be raised against, especially as I did not ever used austin.

```
ssbarnea@m1: ~/c/a/debops master ⚡
$ austin --version
austin 3.3.0
 
ssbarnea@m1: ~/c/a/debops master ⚡
$ austin-tui python --version
🏁 Starting the Austin TUI ...❌ Austin failed to start:                                                    

  ❯ Austin did not start properly

Please make sure that the Austin binary is available from the PATH environment
variable and that the command line arguments that you have provided are correct.
FAIL: 255
```

The same error appears regardless what arguments I give to python.

**Reply by P403n1x87** [reply]
On MacOS, Austin and any tools that use it must be invoked with `sudo`. Trying to profile a command like `python --version` might still fail though, since the program is very short-lived and Austin won't have the chance to profile it. Something like
~~~ console
sudo austin-tui python -c "from time import sleep; sleep(3)"
~~~
should work fine though.
<img width="912" alt="Screenshot 2022-04-29 at 10 27 35" src="https://user-images.githubusercontent.com/20231758/165919083-9aebe6c0-1b75-4c15-a2c1-657ce1605ca9.png">

**Reply by ssbarnea** [reply]
Lol, it worked. I kinda feel bit stupid now. I wonder if it is possible to make it work without sudo, on macos i barely use sudo and I am really scared about what it could do.

**Reply by ssbarnea** [reply]
One problem that I see is that it does not work as non-root and the error message gives does not give any indication than `sudo` is needed. Mentioning it could help others.

**Reply by P403n1x87** [reply]
Unfortunately, tools like Austin really need to run as `sudo` on MacOS. The alternatives are probably worse (e.g. disabling SIP system-wide). If you use Docker on MacOS, you could then run Austin there.

Good shout on improving the error message from `austin-tui`. I'll see what I can do about that!

**Reply by ssbarnea** [reply]
The old `sys_platform == 'darwin'` check could help here.
